src/lib/accounts/index.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  AccountBase,
+  sign,
+  messageToHash,
+  RpcRejectedByUserError,
+  unpackTx,
+} from '@aeternity/aepp-sdk-13';
+import { Encoded } from '@aeternity/aepp-sdk-13/src/utils/encoder';
+import { Store } from 'vuex';
+import { useAccounts } from '../../composables/accounts';
+import { IS_CORDOVA, IS_EXTENSION_BACKGROUND } from '../environment';
+import { POPUP_TYPE_MESSAGE_SIGN, POPUP_TYPE_SIGN, POPUP_TYPE_TX_SIGN } from '../../popup/utils';
+import { showPopup } from '../../background/popupHandler';
+
+export class AccountSuperhero extends AccountBase {
+  address: Encoded.AccountAddress;
+
+  store:Store<any>;
+
+  aeppInfo: Record<string, any> = {};
+
+  constructor(store: Store<any>) {
+    super();
+    this.store = store;
+    const { account } = useAccounts({ store: this.store });
+    this.address = account.value.address as Encoded.AccountAddress;
+  }
+
+signTransaction = (txBase64: Encoded.Transaction, opt: any): Promise<Encoded.Transaction> => {
+  if (IS_CORDOVA) {
+    return this.fgPermissionCheckAndSign(txBase64, opt, opt.aeppOrigin);
+  }
+  if (IS_EXTENSION_BACKGROUND) {
+    return this.bgPermissionCheckAndSign(
+      POPUP_TYPE_TX_SIGN,
+      txBase64,
+      { ...opt, origin: opt.aeppOrigin },
+    );
+  }
+  return this.store.dispatch('accounts/signTransaction', { txBase64, opt: { ...opt, onAccount: this.address } });
+}
+
+signMessage = (message: string, opt: any): Promise<Uint8Array> => {
+  if (IS_CORDOVA) {
+    return this.fgPermissionCheckAndSign(messageToHash(message), opt, opt.aeppOrigin);
+  }
+  if (IS_EXTENSION_BACKGROUND) {
+    return this.bgPermissionCheckAndSign('message.sign', message, { ...opt, origin: opt.aeppOrigin });
+  }
+  return this.sign(messageToHash(message), opt);
+}
+
+sign = (data: string | Uint8Array, opt: any): Promise<Uint8Array> => IS_EXTENSION_BACKGROUND
+  ? sign(data, this.store.getters.account.secretKey) as any
+  : this.store.dispatch('accounts/sign', data, opt)
+
+fgPermissionCheckAndSign = async (payload: any, opt: any, origin: any = null) => {
+  try {
+    const host = origin ? new URL(origin).host : null;
+    const permission = (!host && IS_CORDOVA) || await this.store.dispatch('permissions/checkPermissions', {
+      host,
+      method: POPUP_TYPE_SIGN,
+      params: payload,
+    });
+
+    return this.store.dispatch('accounts/signTransaction', {
+      txBase64: payload,
+      opt: {
+        ...opt,
+        modal: !permission,
+        host,
+      },
+    });
+  } catch (error) {
+    throw new RpcRejectedByUserError(error.message);
+  }
+};
```

Comment:
Aren't you missing a two spaces at the beginning here?

Reply:
Good catch.